packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,204 @@
+
+  /**
+   * The item ID
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: id
+   */
+  @attr
+  public id: string = uniqueId('tooltip-');
+
+  /**
+   * Set the delay for the tooltip
+   */
+  @attr({ converter: nullableNumberConverter })
+  public delay?: number;
+
+  /**
+   * The default delay for the tooltip
+   * @internal
+   */
+  private defaultDelay: number = 250;
+
+  /**
+   * Set the positioning of the tooltip
+   */
+  @attr
+  public positioning?: string;
+
+  /**
+   * The id of the anchor element for the tooltip
+   */
+  @attr
+  public anchor: string = '';
+
+  /**
+   * Reference to the anchor element
+   * @internal
+   */
+  private get anchorElement(): HTMLElement | null {
+    const rootNode = this.getRootNode();
+    return (rootNode instanceof ShadowRoot ? rootNode : document).getElementById(this.anchor ?? '');
+  }
+
+  /**
+   * Reference to the anchor positioning style element
+   * @internal
+   */
+  protected anchorPositioningStyleElement: HTMLStyleElement | null = null;
+
+  public constructor() {
+    super();
+    this.elementInternals.role = 'tooltip';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+
+    // If the anchor element is not found, tooltip will not be shown
+    if (!this.anchorElement) {
+      return;
+    }
+
+    // @ts-expect-error - Baseline 2024
+    const anchorName = this.anchorElement.style.anchorName || `--${this.anchor}`;
+
+    const describedBy = this.anchorElement.getAttribute('aria-describedby');
+    this.anchorElement.setAttribute('aria-describedby', describedBy ? `${describedBy} ${this.id}` : this.id);
+
+    if (SUPPORTS_ANCHOR_POSITIONING) {
+      // @ts-expect-error - Baseline 2024
+      this.anchorElement.style.anchorName = anchorName;
+      // @ts-expect-error - Baseline 2024
+      this.style.positionAnchor = anchorName;
+    } else {
+      // Provide style fallback for browsers that do not support anchor positioning
+      if (!this.anchorPositioningStyleElement) {
+        this.anchorPositioningStyleElement = document.createElement('style');
+        document.head.append(this.anchorPositioningStyleElement);
+      }
+
+      // Given a position with <direction>-<alignment> format, return the proper CSS properties
+      // eslint-disable-next-line prefer-const
+      let [direction, alignment] = this.positioning?.split('-') ?? [];
+
+      if (alignment === undefined && (direction === 'above' || direction === 'below')) {
+        alignment = 'centerX';
+      }
+      if (alignment === undefined && (direction === 'before' || direction === 'after')) {
+        alignment = 'centerY';
+      }
+
+      const directionCSS =
+        {
+          above: `bottom: anchor(${anchorName} top);`,
+          below: `top: anchor(${anchorName} bottom);`,
+          before: `right: anchor(${anchorName} left);`,
+          after: `left: anchor(${anchorName} right);`,
+        }[direction] || 'above';
```

Comment:
Should `above` here be `bottom: anchor(${anchorName} top);`?

Reply:
Shoot. Thanks. I had it inside the key `[direction || 'above']` as the fallback... but that felt weird. Willfix.

Reply:
Yeah. I don't think `[direction || 'above']` would work either since`direction` can be string but not a valid key.

Reply:
This should be fixed up now. Please take a look.

Reply:
This should do it!